pkg/driver/controller_modify_volume.go

```diff
@@ -151,7 +151,9 @@ func executeModifyVolumeRequest(c cloud.Cloud) func(string, modifyVolumeRequest)
 			if err != nil {
 				switch {
 				case errors.Is(err, cloud.ErrInvalidArgument):
-					return 0, status.Errorf(codes.InvalidArgument, "Could not modify volume (invalid argument) %q: %v", volumeID, err)
+					// Returning Internal error instead of InvaliArgument because at this point any tag modifications have succeeded.
+					// It would not be correct to return an error that is considered infeasible by the resizer if the volume was already modified in any way.
+					return 0, status.Errorf(codes.Internal, "Could not modify volume (invalid argument) %q: %v", volumeID, err)
```

Comment:
Offline discussion notwithstanding, I think we should only return `Internal` here if a tag modification was actually made, so we should probably pass that into this function and branch the return code based on that.